### DISCUSSION TITLE
Internal differences between component and render function

### DISCUSSION THREAD

**Discussion by maxmezzomo** [solidjs/solid#1395] [Q&A] [answered]
Hi great framework lots of fun to use! thank you.
I'm still figuring out how the reactivity works, more understand in theory but different in practice.
Specifically wondering what the practical differences would be between creating a component:

```
const Button = (props) => <button>{props.text}</button>
… <App> <Button text="hi"/> </App>
```
and a render function
```
const renderButton = (props) => <button>{props.text}</button>
… <App> {renderButton({text: "hi"})} </App>

//or even 
const renderButton = (props, someOtherContext) => <button onClick={()=>someOtherContext.doSomething()}>
{props.text}</button>
… <App> {renderButton({text: "hi"},{doSomething:()=>{ }})} </App>
```

Within React, I could tell the difference and could use render functions where appropriate, when we essentially did not need any state so just render logic. In Solid the same approach is working and I guess it is because the function, being a simple function, is using whatever scope it was declared in. Still just curious if there are any considerations regarding doing this.

Thank you.

**Comment by ryansolid** [comment, marked as answer]
Components are run through `untrack`. It helps isolate reactivity by not tracking in the parent. That's basically it. Expressions are wrapped anyway so it just protects from things like top-level reads or accidental destructuring triggering unintended things above.
